File: proto-java/src/main/java/com/spaceape/protobuf/reflect/BoolFieldDescriptor.java
package com.spaceape.protobuf.reflect;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.spaceape.protobuf.CodeGenContext;

import java.io.IOException;

public class BoolFieldDescriptor extends FieldDescriptor{
    public BoolFieldDescriptor(int fieldNumber, String fieldName, Class clazz){
        super(fieldNumber, fieldName, clazz, boolean.class);
    }
    public Boolean readField(CodedInputStream input, CodeGenContext context) throws IOException {
        return input.readBool();
    }
    public void writeField(CodedOutputStream output, CodeGenContext context, Object object) throws IOException{
        output.writeBool(fieldNumber, (boolean)object);
    }
}

File: proto-csharp/ProtoMessage/ProtoMessage/ApeMessage/Reflect/BoolFieldDescriptor.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Google.ProtocolBuffers;
using com.spaceape.protobuf;

namespace com.spaceape.protobuf.reflect
{
  public class BoolFieldDescriptor: FieldDescriptor
  {
    public BoolFieldDescriptor(int fieldNumber, String fieldName, Type clazz) : base(fieldNumber, fieldName, clazz)
    {
    }

    override public Object readField(CodedInputStream input, CodeGenContext context)
    {
      return input.ReadBool();
    }

    override public void writeField(CodedOutputStream output, CodeGenContext context, Object obj)
    {
      output.WriteBool(fieldNumber, (bool)obj);
    }
  }
}

